### PR TITLE
Add section on disabling scaladoc

### PR DESCRIPTION
**Pull request by majk-p**
Disabling scaladoc is known to improve build times for large codebases (example: https://github.com/scala/scala3/pull/18157). Multiple projects do it (example https://github.com/scala/scala/blob/2.13.x/build.sbt#L362) but it doesn't seem to be documented. Hopefully that's useful.

Also please let me know if there are other/better options to do it.

### CODE REVIEW

**Comment by majk-p** [comment]
Doing some more search over Github repositories, I noticed combined use of  `Compile / doc / skip := true` along with emptying `Compile / doc / sources`. Should both be referenced in the docs?

**Comment by eed3si9n** [comment]
I don't know if `Compile / doc / skip` does anything, so I'd recommend against that unless it's proven to be useful.

**Comment by eed3si9n** [comment]
> Disabling scaladoc is known to improve build times for large codebases

What does "build times" mean here? I don't think `compile` or `test` would invoke `doc`. Do you mean for `publishLocal`?

**Comment by majk-p** [comment]
> Do you mean for publishLocal?

Yes, in my quite specific case I need to run `publishLocal` against some of my generated sources to run `scripted` tests. Do you think it's better to reword it to something like `reduce the time consumed by publishing task`?

**Comment by majk-p** [comment]
CLA signed
